Code/GraphMol/SmilesParse/CXSmilesOps.cpp

```diff
@@ -755,18 +754,19 @@ void processCXSmilesLabels(RDKit::RWMol &mol) {
         this is a Marvin internal thing and just parse it as they describe it.
         This means that "*" in the SMILES itself needs to be treated
         differently, which we do below. */
-        addquery(makeAHAtomQuery(), symb, mol, (*atIt)->getIdx());
+        addquery(makeAHAtomQuery(), symb, mol, atom->getIdx());
       } else if (symb == "X_p") {
-        addquery(makeXAtomQuery(), symb, mol, (*atIt)->getIdx());
+        addquery(makeXAtomQuery(), symb, mol, atom->getIdx());
       } else if (symb == "XH_p") {
-        addquery(makeXHAtomQuery(), symb, mol, (*atIt)->getIdx());
+        addquery(makeXHAtomQuery(), symb, mol, atom->getIdx());
       } else if (symb == "M_p") {
-        addquery(makeMAtomQuery(), symb, mol, (*atIt)->getIdx());
+        addquery(makeMAtomQuery(), symb, mol, atom->getIdx());
       } else if (symb == "MH_p") {
-        addquery(makeMHAtomQuery(), symb, mol, (*atIt)->getIdx());
+        addquery(makeMHAtomQuery(), symb, mol, atom->getIdx());
       }
-    } else if ((*atIt)->getAtomicNum() == 0 && (*atIt)->getSymbol() == "*") {
-      addquery(makeAAtomQuery(), "", mol, (*atIt)->getIdx());
+    } else if (atom->getAtomicNum() == 0 && !atom->hasQuery() &&
+               atom->getSymbol() == "*") {
```

Comment:
It looks like the only difference here was checking whether the atom already has a query, is this correct?

I admit to being confused about the logic, is the cxsmarts adding the dummy atom query but then we need to ignore it for the output cxsmarts purposes?

Reply:
This function is primarily there to handle atom labels ilke "Q", "X", etc., which can also show up in CXSMILES and which need to be converted to the appropriate query. When you have "A" (=any heavy atom) in a molecule and convert it to CXSMILES in Marvin Sketch you end up with just a dummy atom in the CXSMILES and no additional marker. This logic is there to cover that case.

Here's an example, drawing this:
![image](https://user-images.githubusercontent.com/540511/128282298-c36bc5f5-90a2-47d6-bdc4-e545ca6841b9.png)
yields this CXSMILES: `**-* |$Q_e;;star_e$|`
which the RDKit parses to:
```
>>> rdkit.__version__
'2021.03.4'
>>> Chem.MolToSmarts(Chem.MolFromSmiles('**-* |$Q_e;;star_e$|'))
'[!#6&!#1]-[!#1]-*'
```

The logic that I added in this PR, which will only end up applying when parsing SMARTS, makes sure that logic doesn't get applied to atoms which come from SMARTS, where some query features end up matching the original logic.